jgrapht-core/src/main/java/org/jgrapht/Graph.java

```diff
@@ -576,5 +576,21 @@
      */
     void setEdgeWeight(E e, double weight);
 
+    /**
+     * Assigns a weight to an edge between <code>sourceVertex</code> and <code>targetVertex</code>.
+     * If no edge exists between <code>sourceVertex</code> and <code>targetVertex</code> or either of these vertices is
+     * <code>null</code>, a <code>NullPointerException</code> is thrown.
+     * <p> When there exist multiple edges between <code>sourceVertex</code> and <code>targetVertex</code>, consider using
```

Comment:
Needs some more documentation. What happens if there are multiple edges between source and target. Do they all change weights or just the first one.